jsonschema/validators.py

```diff
@@ -862,6 +864,40 @@ def resolve_remote(self, uri):
             self.store[uri] = result
         return result
 
+    def store_subschema(self, schema, last_url=None):
+        """
+        Using $id or id with $ref, save subschema to self.store
+
+        Arguments:
+
+            schema:
+
+                The referring schema.
+
+            last_url:
+
+                The last URL.
+        """
+        if not isinstance(schema, dict):
```

Comment:
Will have a closer look, but the schema does not have to be a dict (it's an [arbitrary collections.Mapping](https://github.com/Julian/jsonschema/blob/main/jsonschema/validators.py#L933)) and does not necessarily even need to be copy-able.